core/state.go

```diff
@@ -13,78 +13,80 @@ import (
 
 // Operators
 
-// OperatorSocket is formatted as "host:dispersalPort;retrievalPort;v2DispersalPort"
-type OperatorSocket string
+// OperatorSocket is formatted as "host:dispersalPort;retrievalPort;v2DispersalPort;v2RetrievalPort"
+type OperatorSocket struct {
+	host            string
+	v1DispersalPort string
+	v1RetrievalPort string
+	v2DispersalPort string
+	v2RetrievalPort string
+}
 
-func (s OperatorSocket) String() string {
-	return string(s)
+// NewOperatorSocket returns new OperatorSocket struct
+func NewOperatorSocket(nodeIP, dispersalPort, retrievalPort, v2DispersalPort, v2RetrievalPort string) OperatorSocket {
+	return OperatorSocket{
+		host:            nodeIP,
+		v1DispersalPort: dispersalPort,
+		v1RetrievalPort: retrievalPort,
+		v2DispersalPort: v2DispersalPort,
+		v2RetrievalPort: v2RetrievalPort,
+	}
 }
 
-func MakeOperatorSocket(nodeIP, dispersalPort, retrievalPort, v2DispersalPort, v2RetrievalPort string) OperatorSocket {
-	//TODO:  Add config checks for invalid v1/v2 configs -- for v1, both v2 ports must be empty and for v2, both ports must be valid, reject any other combinations.
-	if v2DispersalPort == "" && v2RetrievalPort == "" {
-		return OperatorSocket(fmt.Sprintf("%s:%s;%s", nodeIP, dispersalPort, retrievalPort))
+func (s OperatorSocket) Normalize() string {
```

Comment:
Given the opposite function is `ParseOperatorSocket`, it may be clear to use something like `Encode()` (Parse/Encode pair is clear)?